### DISCUSSION TITLE
How do I register React Flags Select with React Hook Form.

### DISCUSSION THREAD

**Discussion by primado** [react-hook-form/react-hook-form#9958] [Q&A] [answered]
` <ReactFlagsSelect
                                            selected={selected}
                                            onSelect={code => setSelected(code)}
                                            searchPlaceholder="Search for a country"
                                            fullWidth={true}
                                            searchable={true}
                                            {...register("flagSelect", {required: {
                                                value: true,
                                                message: "Nationality is required"
                                            }})}
                                            className="font-nunito font-medium text-lg w-80 bg-white outline-none rounded-lg focus:border-main-green" />
          { errors.flagSelect && <p className="text-red-500 text-sm font-medium">{ errors.flagSelect.message }</p>}`

**Comment by dbrxnds** [comment]
I think you will need to use `Controller`  component from react-hook-form for this

See: https://react-hook-form.com/api/usecontroller/controller/

**Reply by primado** [reply]
Yes, I tried to use Controller but anytime I select a country, the error message pops up saying Nationality is required even after selecting.

**Comment by Moshyfawn** [comment, marked as answer]
It's hard to tell what's wrong without a repoduction sandbox but[ here's my answer](https://github.com/react-hook-form/react-hook-form/discussions/8007#discussioncomment-2350446) to a different thread about RFS + RHF integration with [a live example](https://codesandbox.io/s/modern-browser-udlqkx?file=/src/App.tsx).

**Reply by primado** [reply]
I have seen how you did it and I have applied it to mine.
Thanks

**Reply by primado** [reply]
@Moshyfawn how can I write the default value for React Flag Select using RHF. I tried but it didn't work. 
Thank you

**Reply by primado** [reply]
I'm using JavaScript but not TypeScript

**Reply by Moshyfawn** [reply]
I'm not sure what you mean. In [the example I provided](https://codesandbox.io/s/modern-browser-udlqkx?file=/src/App.tsx) RFS default value is set via `defaultValues` and passed into the `useForm` hook.
```js
const defaultValues = {
  // INFO: assuming your input name is "country"
  country: "US"
}

useForm({ defaultValues })
```
